module/apmfasthttp/context.go

```diff
@@ -111,7 +109,7 @@ func StartTransactionWithBody(
 		return nil, nil, err
 	}
 
-	ctx.SetUserValue(txKey, newTxCloser(tx, bc))
+	ctx.SetUserValue(txKey, newTxCloser(ctx, tx, bc))
```

Comment:
The fasthttp docs state:

> VERY IMPORTANT! Fasthttp disallows holding references to RequestCtx or to its' members after returning from RequestHandler. Otherwise data races are inevitable. Carefully inspect all the net/http request handlers converted to fasthttp whether they retain references to RequestCtx or to its' members after returning.

With your change, isn't StartTransactionWithBody now causing a reference to RequestCtx to be held (in ctx itself) after the handler returns?

Reply:
The docs reference to use a RequestCtx after the RequestHandler finish (parallel gorutines that use the RequestCtx), but the context user values are allowed, because it's reset before being put back into the pool. So it's safe!

Reply:
Sure! 😉